form-view-model/src/main/kotlin/com/ritense/formviewmodel/web/rest/error/FormViewModelModuleExceptionTranslator.kt

```diff
@@ -0,0 +1,39 @@
+package com.ritense.formviewmodel.web.rest.error
+
+import com.ritense.formviewmodel.error.BusinessException
+import com.ritense.formviewmodel.error.FormException
+import com.ritense.formviewmodel.web.rest.dto.BusinessRuleError
+import com.ritense.formviewmodel.web.rest.dto.FormError
+import org.springframework.core.Ordered
+import org.springframework.core.annotation.Order
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.ControllerAdvice
+import org.springframework.web.bind.annotation.ExceptionHandler
+import org.springframework.web.context.request.NativeWebRequest
+import org.zalando.problem.spring.web.advice.ProblemHandling
+
+@ControllerAdvice
+@Order(Ordered.HIGHEST_PRECEDENCE)
+class FormViewModelModuleExceptionTranslator : ProblemHandling {
+
+    @ExceptionHandler(FormException::class)
+    fun handleFormException(
+        ex: FormException,
+        request: NativeWebRequest
+    ): ResponseEntity<FormError> {
+        return ResponseEntity
+            .badRequest()
+            .body(FormError(error = ex.message!!, component = ex.component))
```

Comment:
I think it's nicer if the exception translator builds up the FormError. The FormError is just a DTO and I'd rather not have any logic in there

Reply:
That is a little bit difficult since `String` and `String?` have the same signature. It's also still putting in logic in a DTO which I'm not a fan of. 

What if I create a companion method instead to build the object from an Exception instead? The logic won't be completely part of the DTO that way, but they are still not very separated either

Reply:
I ended up doing this through a builder, that way the `Unknown Form Error` can still be consistent but without any logic in the DTO